tests/OAuth2ClientHandler.Tests/Helpers/Startup.cs

```diff
@@ -22,7 +22,8 @@ public void Configuration(IAppBuilder app)
                         OnValidateClientAuthentication = (ctx) =>
                         {
                             string clientId, clientSecret;
-                            if (ctx.TryGetBasicCredentials(out clientId, out clientSecret) &&
+                            if ((ctx.TryGetBasicCredentials(out clientId, out clientSecret) ||
+                                 ctx.TryGetFormCredentials(out clientId, out clientSecret)) &&
```

Comment:
I've created another method instead to make the code more clear. Done in 07c5f84.